src/NzbDrone.Core/ImportLists/AniList/AniListSettingsBase.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using FluentValidation;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Annotations;
+using NzbDrone.Core.Validation;
+
+namespace NzbDrone.Core.ImportLists.AniList
+{
+    public class AniListSettingsBaseValidator<TSettings> : AbstractValidator<TSettings>
+    where TSettings : AniListSettingsBase<TSettings>
+    {
+        public AniListSettingsBaseValidator()
+        {
+            RuleFor(c => c.BaseUrl).ValidRootUrl();
+
+            RuleFor(c => c.AccessToken).NotEmpty()
+                                       .OverridePropertyName("SignIn")
+                                       .WithMessage("Must authenticate with AniList");
+
+            RuleFor(c => c.RefreshToken).NotEmpty()
+                                        .OverridePropertyName("SignIn")
+                                        .WithMessage("Must authenticate with AniList")
+                                        .When(c => c.AccessToken.IsNotNullOrWhiteSpace());
+
+            RuleFor(c => c.Expires).NotEmpty()
+                                   .OverridePropertyName("SignIn")
+                                   .WithMessage("Must authenticate with AniList")
+                                   .When(c => c.AccessToken.IsNotNullOrWhiteSpace() && c.RefreshToken.IsNotNullOrWhiteSpace());
+        }
+    }
+
+    public class AniListSettingsBase<TSettings> : IImportListSettings
+        where TSettings : AniListSettingsBase<TSettings>
+    {
+        protected virtual AbstractValidator<TSettings> Validator => new AniListSettingsBaseValidator<TSettings>();
+
+        public AniListSettingsBase()
+        {
+            BaseUrl = "https://graphql.anilist.co";
+            SignIn = "startOAuth";
+            MapSourceUrl = "https://raw.githubusercontent.com/Fribb/anime-lists/master/anime-list-full.json";
```

Comment:
There seems to be a few of these lists available, but in slightly different formats. I wasn't sure what the best approach here, but as it stands currently a setting does seem silly. Originally, i also had the source and destination keys as settings as well, that way it could be changed on the fly, but not every source is a flat list, so I ended up backing it off.

As you say, a service would be better in order to separate this specific data parsing from the importer itself, especially with multiple imports now in play.

I am assuming that will be a later refactor, though, and if so, the setting here can be removed at that time?

Reply:
I need to look at what it's going to need on the Skyhook side, but regardless I think the setting should be removed.

Reply:
I removed the setting as requested.

I would like to point out, however, that BaseUrl is also a setting that doesn't really make sense in the same context. However, it is an existing pattern.